device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -219,6 +219,11 @@ void setMaxNumberOfMessagesSentPerSendThread(int maxNumberOfMessagesSentPerSendT
      */
     private void startWorkerThreads()
     {
+        // while startWorkerThreads should never be called when threads are already active, it doesn't hurt to double
+        // check that any previous thread pools have been shut down.
+        stopWorkerThreads();
```

Comment:
While not necessary, I figured it didn't hurt to add this check here just in case. When protecting against thread leaking issues, you can't be too careful